### ISSUE TITLE
PhantomJS tests fail on detectChanges

### ISSUE THREAD

**Issue by simplyspoke** [500tech/angular-tree-component#433]
I have been able to successfully implement a version of the tree component that has tri-state check-boxes, however there seems to be an uncaught error that occurs when the test runs `fixture.detectChanges()`. 

Here is the error I am getting:

```
ERROR: '[mobx] Encountered an uncaught exception that was thrown by a reaction or observer component, in: 'Reaction[TreeNodeCollectionComponent.detectChanges()]', TypeError{line: 44710, sourceURL: 'http://localhost:9876/_karma_webpack_/vendor.bundle.js', stack: 'iterateListLike@http://localhost:9876/_karma_webpack_/vendor.bundle.js:44710:39
```

Has anyone else been able to successfully run tests in PhantomJS?

**Comment by adamkleingit** [comment]
Never tested the tree in PhantomJS. I would suggest to use a different browser for testing, or to mock the tree component when running in PhantomJS.

**Comment by hpawe01** [comment]
I don't see, how this issue is resolved. Sometimes we need to use a headless browser like PhantomJS, e.g. when running automated tests on the CI-Server.

I also get an error, when running `fixture.detectChanges()`:
```
TypeError: undefined is not a function (evaluating 'nodes.forEach') in http://localhost:9876/_karma_webpack_/vendor.bundle.js (line 1684)
```

What do you suggest to mock the component? `ng2-mock-component`?

**Comment by simplyspoke** [comment]
We adjusted in a couple of ways.

First, by shallow testing the component that we were using the tree on, we ensure that we we not testing the code of others. That seems to be a dangerous direction to head if you are looking to maximize the value of your work.

Secondly, we shifted out tests to use chrome headless instead of PhantomJS. Phantom has been a great tool over the years, but many new projects have closed the gaps that it filled and it has been appropriately abandoned.

Best to shift to a more modern headless browser and not invest your time in issues with Phantom.
